ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -2118,3 +2095,51 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
         self.assertEqual(suborder_data['Material Type'], material_type)
         self.assertEqual(suborder_data['Article Name'], article)
         self.assertEqual(suborder_data['Test Units'], test_unit)
+
+    def test034_duplicate_sub_order_table_with_add(self):
+        """
+        Orders: User can duplicate any suborder from the order form ( table with add )
+        LIMS-3738
+        :return:
+        """
+        # get random order
+        self.base_selenium.LOGGER.info('select random record')
+        orders, payload = self.orders_api.get_all_orders(limit=20)
+        data_before_duplicate_main_order = random.choice(orders['orders'])
+
+        #filter by this random order
+        order_no = data_before_duplicate_main_order['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no,
+                                              field_type='text')
+        #get order data before duplicate
+        row = self.order_page.get_last_order_row()
+        self.orders_page.open_edit_page(row)
+        data_before_duplicate = self.order_page.get_suborder_data()
+
+        self.orders_page.get_orders_page()
```

Comment:
why do you need to repeat that steps  from line 2219 to 2126 again ?

Reply:
@ghada-37 because I want to get sub order data before duplicate and after the duplication

Reply:
No need to go back to order page and filter and open edit page you already opened it all you need is to :
1) open edit page of random order
2) get suborder data before duplicate
3)duplicate it 
4) get suborder data after duplicate

Reply:
@dianaeldemerdash Could u check this comment?